assets/src/dashboard/animations/parts/blinkOn/index.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import { ANIMATION_TYPES } from '../../constants';
+import SimpleAnimation from '../simpleAnimation';
+import generateLookupMap from '../../utils/generateLookupMap';
+import padArray from '../../utils/padArray';
+
+const AVAILABLE_OPACITY = [0, 0.25, 0.75, 1];
+const DEFAULT_BLINKS = 10;
+
+const defaults = {
+  fill: 'forwards',
+};
+
+function generateOpacityFrames(count) {
+  const blinkCount = count || DEFAULT_BLINKS;
+  const opacityFrames = [];
+  const lookup = generateLookupMap(AVAILABLE_OPACITY);
+
+  for (let i = 0; i < blinkCount - 2; i++) {
+    const opacities = i > 0 ? lookup[opacityFrames[i - 1]] : AVAILABLE_OPACITY;
+    const opacity = opacities[Math.floor(Math.random() * opacities.length)];
+
+    opacityFrames.push(opacity);
+  }
+
+  return [...padArray(opacityFrames, 3)];
+}
+
+export function AnimationBlinkOn({ blinkCount, ...args }) {
+  const timings = {
+    ...defaults,
+    ...args,
+  };
+
+  const animationName = `${uuidv4()}-${ANIMATION_TYPES.BLINK_ON}`;
+  const keyframes = {
+    opacity: [0, ...generateOpacityFrames(blinkCount), 1],
+  };
+
+  const { id, WAAPIAnimation, AMPAnimation, AMPAnimator } = SimpleAnimation(
+    animationName,
+    keyframes,
+    timings
+  );
+
+  return {
+    id,
+    WAAPIAnimation,
+    AMPAnimation,
+    AMPAnimator,
```

Comment:
Maybe a little late for this, but I do feel the concept of `Animation` & `Animator` is confusing, especially because the `Animator` here performs the same function as a Web Animations API's `Animation`. (https://developer.mozilla.org/en-US/docs/Web/API/Animation)

I guess what we really have here are drivers and targets. For amp you're separating out the driver and target on creation where as for WAAPI the target is creating the driver and communicating it to the provider on render.

Also the `Animation` here is just a wrapper that allows the driver to target it.

Not saying we have to change it, but I do think this would gain some clarity if it was:
```
    WAAPITarget,
    AMPTarget,
    AMPAnimation,
```
or
```
    WAAPIWrapper,
    AMPWrapper,
    AMPAnimation,
```
only because I feel like calling the driver an `Animator` and the target an `Animation` makes it counter intuitive because it's assigning the same name a native API has established (Animation) to a different responsibility.

Reply:
I understand this would be a sweeping change, so probably don't want to do it. Just have a mental hiccup everytime I see the existing names